### PR TITLE
fix: Modify end_to_end.yml to be more specific about which gtfs-validator jar is used

### PR DESCRIPTION
**Pull request by bdferris-v2**
Per discussion in issue #1129, I think I ran into an edge-case where there are multiple jars in the main/build/libs directory (maybe from a previous build?) that caused the script to fail.  Attempting to fix by being more specific about which of the :main project jars to use (hint: we want the shadow jar).

### CODE REVIEW

**Review by barbeau — COMMENTED** [review]
These changes LGTM, although the same pattern exists in end_to_end_big.yml and end_to_end_100.yml as well. I'd suggest we change all these occurrences to be consistent.

**Comment by bdferris-v2** [comment]
@barbeau good call.  Done.

**Review by barbeau — APPROVED** [review]
Thanks @bdferris-v2, LGTM!

The acceptance tests are still blocked by https://github.com/MobilityData/gtfs-validator/issues/1094, so if we want them to run before merge this PR would need to be merged after PR https://github.com/MobilityData/gtfs-validator/pull/1102.

**Comment by bdferris-v2** [comment]
All the end_to_end*.yml tests look to be passing, which I think is the most important thing to check here.  So I propose to submit without waiting for #1102.  I will do so this evening unless I hear objections.

**Comment by bdferris-v2** [comment]
But maybe you were instead pointing out that I won't be able to submit this PR until #1102 is in, regardless?

**Comment by barbeau** [comment]
No objections here. I think the repo permissions allow merge even with failing/canceled tests.

**Comment by maximearmstrong** [comment]
Thank you for this contribution @bdferris-v2 and @barbeau ! We didn’t get a chance to add this to the contribution guidelines yet, but can you make sure there is a review from one MobilityData staff before merging please? :slightly_smiling_face:

cc @isabelle-dr

**Pull request merged**
